File: schemaorg-rs/src/program_membership.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type ProgramMembershipAdditionalTypeFieldEnum = String;
///<https://schema.org/ProgramMembership>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct ProgramMembership {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/membershipPointsEarned>
    #[serde(rename = "membershipPointsEarned")]
    #[serde_as(as = "OneOrMany<_>")]
    pub membership_points_earned: Vec<ProgramMembershipMembershipPointsEarnedFieldEnum>,
    ///<https://schema.org/hostingOrganization>
    #[serde(rename = "hostingOrganization")]
    #[serde_as(as = "OneOrMany<_>")]
    pub hosting_organization: Vec<Organization>,
    ///<https://schema.org/membershipNumber>
    #[serde(rename = "membershipNumber")]
    #[serde_as(as = "OneOrMany<_>")]
    pub membership_number: Vec<String>,
    ///<https://schema.org/programName>
    #[serde(rename = "programName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub program_name: Vec<String>,
    ///<https://schema.org/members>
    #[serde(rename = "members")]
    #[serde_as(as = "OneOrMany<_>")]
    pub members: Vec<ProgramMembershipMembersFieldEnum>,
    ///<https://schema.org/member>
    #[serde(rename = "member")]
    #[serde_as(as = "OneOrMany<_>")]
    pub member: Vec<ProgramMembershipMemberFieldEnum>,
    ///<https://schema.org/program>
    #[serde(rename = "program")]
    #[serde_as(as = "OneOrMany<_>")]
    pub program: Vec<MemberProgram>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<ProgramMembershipMainEntityOfPageFieldEnum>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<ProgramMembershipIdentifierFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<ProgramMembershipDescriptionFieldEnum>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<ProgramMembershipSubjectOfFieldEnum>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<ProgramMembershipAdditionalTypeFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<ProgramMembershipImageFieldEnum>,
}

File: schemaorg-ts/src/program_membership.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type ProgramMembershipAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/ProgramMembership>
 **/
export interface ProgramMembership {
  "@context": string;
  "@type": "ProgramMembership";

  /**
   * <https://schema.org/membershipPointsEarned>
   **/
  membershipPointsEarned: all.OneOrMany<all.ProgramMembershipMembershipPointsEarnedFieldEnum>;

  /**
   * <https://schema.org/hostingOrganization>
   **/
  hostingOrganization: all.OneOrMany<all.Organization>;

  /**
   * <https://schema.org/membershipNumber>
   **/
  membershipNumber: all.OneOrMany<string>;

  /**
   * <https://schema.org/programName>
   **/
  programName: all.OneOrMany<string>;

  /**
   * <https://schema.org/members>
   **/
  members: all.OneOrMany<all.ProgramMembershipMembersFieldEnum>;

  /**
   * <https://schema.org/member>
   **/
  member: all.OneOrMany<all.ProgramMembershipMemberFieldEnum>;

  /**
   * <https://schema.org/program>
   **/
  program: all.OneOrMany<all.MemberProgram>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.ProgramMembershipMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.ProgramMembershipIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.ProgramMembershipDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.ProgramMembershipSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.ProgramMembershipAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.ProgramMembershipImageFieldEnum>;
}
